src/core/server/saved_objects/import/create_saved_objects.ts

```diff
@@ -76,7 +81,70 @@ export const createSavedObjects = async <T>({
   );
 
   // filter out the 'version' field of each object, if it exists
+
   const objectsToCreate = filteredObjects.map(({ version, ...object }) => {
+    if (dataSourceId) {
```

Comment:
Looks like the code is duplicated, can we extract out as a function to be re-used?